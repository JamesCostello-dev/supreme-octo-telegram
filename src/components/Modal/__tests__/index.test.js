import React from "react";
import { render, cleanup, fireEvent } from "@testing-library/react";
import "@testing-library/jest-dom/extend-expect";
import Modal from "..";

const currentPhoto = {
  name: "Park Bench",
  category: "landscape",
  description:
    "Lorem ipsum dolor sit amet, consectetur adipiscing elit. Nunc ultricie",
  index: 1,
};

afterEach(cleanup);

const mockToggleModal = jest.fn();

describe("Modal component", () => {
  it("should render", () => {
    render(<img />);
  });
  it("should render", () => {
    const { asFragment } = render(<img />);
    expect(asFragment()).toMatchSnapshot();
  });
});

describe("Click Event", () => {
  it("should call onClick", () => {
    const { getByText } = render(
      <Modal currentPhoto={currentPhoto} onClose={mockToggleModal} />
    );
    fireEvent.click(getByText("Close this modal"));
    expect(mockToggleModal).toHaveBeenCalledTimes(1);
  });
});
