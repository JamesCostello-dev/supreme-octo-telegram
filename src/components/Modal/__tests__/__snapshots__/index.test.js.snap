// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Modal component should render 1`] = `
<DocumentFragment>
  <img />
</DocumentFragment>
`;
